src/calibre/test_build.py

```diff
@@ -297,11 +297,6 @@ def test_markdown(self):
 
     def test_feedparser(self):
         from calibre.web.feeds.feedparser import parse
-        # sgmllib is needed for feedparser parsing malformed feeds
```

Comment:
I don't see how beautifulsoup 3 to bs4 has anything to do with this line. calibre still very much needs sgmllib.

The test was added in https://github.com/kovidgoyal/calibre/commit/d436e2d7565d1f7ea7c97e0020b16f347e2800e7 and the commit immediately before was when the vendored feedparser (which depends on sgmllib) was removed in-tree.

This test is called test_feedparser, not test_beautifulsoup, and without it we do not know whether feedparser._SGML_AVAILABLE is true so that news recipes will be robust.

Reply:
The test should test something that used by calibre directly, e.g. some functionality of feedparser. Testing for the presence of a third-party module that is not used directly makes assumptions about feedparser dependencies.

Reply:
Okay, so what is your point? Your rationale for removing this is still incorrect, because:
- you're removing something that the author explicitly wanted there, without rationalizing _why_ in a way that answers the initial intent of the test
- your commit message is wrong, it makes an assertion that you are removing the "last references to sgmllib", but the commit you're pointing to never used `sgmllib` in the first place -- it used `calibre.ebooks.sgmllib`

If you feel the test could be better, by testing the existence of functionality rather than assumed indirect dependencies, then maybe you could write a replacement test that tries to parse a malformed feed rather than just removing the import test. :)